### PR TITLE
User stock

### PR DESCRIPTION
**Pull request by JessieFrance**
## What does this PR Do?
This PR aims to:
1. Let users add a stock [#16](https://github.com/chingu-voyages/v37-bears-team-13/issues/16) via a stock symbol. If the stock exists, it's name will be added automatically, and this will become a stock that other users can add as their own personal stock (i.e. associate their account with it).
2. Let users get a list of stocks available in the database that they, or other users have added. The list of stocks here are not associated with specific users (and we should not let regular users be able to directly delete these) but instead are available for anyone signed in to add to their account. 
3. Let users associate a stock with their account so that they can add notes, and so that we can eventually setup the gamification in [#18](https://github.com/chingu-voyages/v37-bears-team-13/issues/18). Users should be able to modify or dis-associate these stocks from their account [#17](https://github.com/chingu-voyages/v37-bears-team-13/issues/17).
4. Start moving the Finnhub API calls to the backend [#70](https://github.com/chingu-voyages/v37-bears-team-13/issues/70).

## How to Test
1. Make sure you add the Finnhub API key to `.env` like in `.env.sample`. 
2. Assuming you are using Port 5000, use Postman to make a Post request to `localhost:5000/api/v1/users/login` so that you are signed in.
3. Make a Post request to `localhost:5000/api/v1/stocks/add-stock` to add a stock with a body field like this: 
```
{
    "symbol": "aapl"
}
```
4. Repeat step 3 with a few more valid and invalid stock symbols. Try sending the same one again.
5. Use Postman to make a GET request to `localhost:5000/api/v1/stocks` to see the full list. Notice that the company names are automatically added.
6. Copy the `id` field on one of the stocks from step 5. In this example, imagine it is `622e4800a84b441b35d0f0ce`.
7. Now associate this stock with your account by making a Post request to `localhost:5000/api/v1/user-stocks/add-stock` with a body of:
```
{
    "stockId": "622e4800a84b441b35d0f0ce",
    "userNotes": "User notes on this stock"
}
```

### CODE REVIEW

**Comment by JessieFrance** [comment]
I forgot to tag that this would help setup some backend logic for [#59](https://github.com/chingu-voyages/v37-bears-team-13/issues/59).

File: src/routes/accounts/signup.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ router.post(
     // Check if user exists in db.
     const existingUser = await User.findOne({ email });
     if (existingUser) {
-      res.status(400).json({ error: 'user already exists' });
+      res.status(409).json({ error: 'user already exists' });
```
Comment (ArturoDucasse) [inline src/routes/accounts/signup.ts:21]: Didn't know that 409 is the correct message error for already saved records.

Thanks for pointing this out

**Review by ArturoDucasse — APPROVED** [review]
Code looks well-formatted, plus possible errors are well handled.

**Review by DanielFleace1 — APPROVED** [review]
This code is very readable and worked in post man and when i was working on my task in the front end. I vote to approve this PR

**Review by rmcguire6 — APPROVED** [review]
Looks good, Jessie. We may need to refactor the routes again when I add Swagger documentation.

**Pull request merged**